dbgpt/storage/graph_store/community_store.py

```diff
@@ -0,0 +1,209 @@
+            Edge objects may include the following attributes:
+
+            src_id: The ID of the source node of the edge
+            dst_id: The ID of the destination node of the edge
+            label: The label or type of the edge
+            properties: A dictionary containing other properties of the edge, e.g., {"description": "some relationship"}
+
+        # community_info example
+        {
+            "cluster1": [
+                "node1 -> node2 -> relationship_type -> relationship description",
+                "node1 -> node3 -> another_relationship -> another description",
+            ],
+            "cluster2": [
+                "node4 -> node5 -> some_relationship -> some description",
+            ],
+        }
+
+        """
+
+        community_info: List[Community] = []
+        tasks = []
+
+        for memory_graph in communities_metadata:
+            for vertex in memory_graph.vertices:
+                task = asyncio.create_task(
+                    self._process_vertex(memory_graph, vertex, community_info)
+                )
+                tasks.append(task)
+
+        await asyncio.gather(*tasks)
+
+        if self._enable_persistence and self._orm:
+            await asyncio.get_event_loop().run_in_executor(
+                self._executor, self._orm.save_community_info, community_info
+            )
+
+        return community_info
+
+    async def _process_vertex(
+        self,
+        memory_graph: Graph,
+        vertex: Vertex,
+        communities: List[Community],
+    ):
+        cluster_id = vertex.properties.get("community_id", "unknown")
+        if cluster_id not in communities:
+            communities[cluster_id] = []
+
+        for edge in memory_graph.edges:
+            if edge.src_id == vertex.id:
+                neighbor_vertex = memory_graph.get_vertex(edge.dst_id)
+                if neighbor_vertex:
+                    detail = f"{vertex.id} -> {neighbor_vertex.id} -> {edge.label} -> {edge.properties.get('description', 'No description')}"
+                    communities[cluster_id].append(detail)
+
+    async def _summarize_communities(self, communities: List[Community]):
+        """Generate and store summaries for each community."""
+        tasks = []
+        for community in communities:
+            task = asyncio.create_task(self._summarize_community(community))
+            tasks.append(task)
+        await asyncio.gather(*tasks)
+
+    async def _summarize_community(self, community: Community):
+        summary = await self._generate_community_summary(community.data)
+        community.summary = summary
+
+        if self._enable_persistence and self._orm:
+            await asyncio.get_event_loop().run_in_executor(
+                self._executor,
+                self._orm.update_community_summary,
+                community.id,
+                summary,
+            )
+
+    async def search_communities(self, query: str) -> List[Community]:
```

Comment:
@M1n9X
The community search function needs to be adapted according to different index structures such as relations and vectors. Here, I am going to add an abstraction api.